### ISSUE TITLE
[ITensors] [BUG] `correlation_matrix` nonsymmetric constraint

### ISSUE THREAD

**Issue by mtfishman** [ITensor/ITensors.jl#810]
@emstoudenmire, the symmetric constraint in `correlation_matrix` seems odd for cases like this:
```julia
using ITensors
using LinearAlgebra
using Random

Random.seed!(1234)

N = 2
m = 2

s = siteinds("S=1/2", N)
psi = randomMPS(s, n -> isodd(n) ? "Up" : "Dn"; linkdims=m)
C = correlation_matrix(psi, "Sz", "Sx")

C_mpo = zeros(N, N)
for i in 1:N, j in 1:N
  os = OpSum()
  os += "Sz", i, "Sx", j
  C_mpo[i, j] = inner(psi, MPO(os, s), psi)
end

println("correlation_matrix")
display(C)

println("\nMPO")
display(C_mpo)
```
which outputs:
```julia
correlation_matrix
2×2 Matrix{Float64}:
 -3.32551e-18   0.234341
  0.234341     -1.45217e-18

MPO
2×2 Matrix{Float64}:
 -9.34059e-19  0.234341
 -0.219799     1.45217e-18
```

Should we have a keyword argument `ishermitian` so that a user can specify that they expect the correlation matrix to be Hermitian? I would say that it should be off by default to be on the safe side, since the symmetric constraint is just an optimization.

Originally pointed out by @JanReimers.

**Comment by mtfishman** [comment]
In https://github.com/ITensor/ITensor/pull/405, in the end this was handled automatically by forming the operators at the beginning and checking if they are Hermitian conjugates of each other (up to some tolerance), in which case the correlation matrix is deduced to be Hermitian. Additionally, users can set a flag to explicitly tell us if they want the correlation matrix to be constrained to be Hermitian or not. That would be a good way to go here as well.

**Comment by mtfishman** [comment]
Closed by #817.
